src/components/motor_output_reordering/MotorOutputReorderingComponent.js

```diff
@@ -0,0 +1,249 @@
+        this._config = new MotorOutputReorderConfig(100);
+
+        this._currentJerkingTimeout = -1;
+        this._currentJerkingMotor = -1;
+
+        this._currentSpinningMotor = -1;
+
+        this._contentDiv.load("./components/motor_output_reordering/body.html", () => {
+            this._setupdialog();
+        });
+    }
+
+    _readDom()
+    {
+        this._domAgreeSafetyCheckBox = $('#motorsEnableTestMode-dialogMotorOutputReorder');
+        this._domAgreeButton = $('#dialogMotorOutputReorderAgreeButton');
+        this._domStartOverButton = $('#motorsRemapDialogStartOver');
+        this._domSaveButton = $('#motorsRemapDialogSave');
+        this._domMainContentBlock = $('#dialogMotorOutputReorderMainContent');
+        this._domWarningContentBlock = $('#dialogMotorOutputReorderWarning');
+        this._domActionHintBlock = $('#motorOutputReorderActionHint');
+        this._domCanvas = $('#motorOutputReorderCanvas');
+    }
+
+    _setupdialog()
+    {
+        i18n.localizePage();
+        this._readDom();
+
+        this._resetGui();
+
+        this._domAgreeSafetyCheckBox.change(() =>
+        {
+            const enabled = this._domAgreeSafetyCheckBox.is(':checked');
+
+            if (enabled) {
+                this._domAgreeButton.show();
+            } else {
+                this._domAgreeButton.hide();
+            }
+        });
+
+        this._domAgreeButton.click(() => {
+            this._onAgreeButtonClicked();
+        });
+        this._domStartOverButton.click(() => {
+            this._startOver();
+        });
+        this._domSaveButton.click(() => {
+            this._save();
+        });
+
+        this._onLoadedCallback();
+    }
+
+    close()
+    {
+        this._stopAnyMotorJerking();
+        this._stopMotor();
+        this._stopUserInteraction();
+        this._resetGui();
+    }
+
+    _resetGui()
+    {
+        this._domMainContentBlock.hide();
+        this._domWarningContentBlock.show();
+        this._domAgreeButton.hide();
+
+        this._domAgreeSafetyCheckBox.prop('checked', false);
+        this._domAgreeSafetyCheckBox.change();
+        this._showSaveStartOverButtons(false);
+    }
+
+    _save() {
+        function save_to_eeprom()
+        {
+            MSP.send_message(MSPCodes.MSP_EEPROM_WRITE, false, false, reboot);
+        }
+
+        function reboot()
+        {
+            GUI.log(i18n.getMessage('configurationEepromSaved'));
+
+            GUI.tab_switch_cleanup(function() {
+                MSP.send_message(MSPCodes.MSP_SET_REBOOT, false, false);
+                reinitialiseConnection(self);
+            });
+        }
+
+        const buffer = [];
+        buffer.push8(this.motorOutputReorderCanvas.readyMotors.length);
+
+        for (let i = 0; i < this._newMotorOutputReorder.length; i++) {
+            buffer.push8(this._newMotorOutputReorder[i]);
+        }
+
+        MSP.send_message(MSPCodes.MSP_SET_MOTOR_OUTPUT_REORDERING, buffer);
```

Comment:
We usually use a variable defined in `fc.js` as intermediate to send/receive the MSP information. I see you are using the buffer directly. Having this intermediate variable helps in a future, when this MSP command grows and can send/receive different data depending on the version.